fontforge/splineutil2.c

```diff
@@ -3654,12 +3654,12 @@ void SplinePointListSet(SplinePointList *tobase, SplinePointList *frombase) {
 	    tsp->noprevcp = fsp->noprevcp;
 	    tsp->nonextcp = fsp->nonextcp;
 	    if ( first==NULL ) first = tsp;
-	    if ( tsp->next==NULL )
+	    if ( tsp->next==NULL || fsp->next==NULL )
```

Comment:
The problem @frank-trampe as I understand it is that the splines get out of sync due to spiros, and soon end up different lengths, an impossible situation with Beziers but perfectly possible with Spiros as the Spiro&rarr;Bezier code guarantees nothing. So this extra sanity check just makes sure we actually have more points to copy from.